### ISSUE TITLE
Teams Presence Sensor

### ISSUE THREAD

**Issue by Clem-** [RogerSelwyn/O365-HomeAssistant#19]
Hi,
Thank you very much for maintaining this great integration!

A [PR ](https://github.com/PTST/O365-HomeAssistant/pull/105/files) is available on the source repo that exposes a Presence sensor based on Microsoft Teams status.
Would it be ok for you to add this feature to your repo?

**Comment by RogerSelwyn** [comment]
I'll take a look. I have no way of testing since I don't use Teams.

**Comment by Clem-** [comment]
I can definitely help you with this. Thanks!

**Comment by RogerSelwyn** [comment]
As written the PR is invalid since the parameter 'conf' on line 42 is undefined. That said, the parameter isn't required in the class called. I'd guess that PR hasn't been tested

**Comment by RogerSelwyn** [comment]
It can be made to work... But will need some work doing to it.... :-(

**Comment by Clem-** [comment]
I got it working by adding `Presence.Read` to `SCOPE` and `MINIMUM_REQUIRED_SCOPES` (and on Azure Portal) and applying [those changes](https://github.com/PTST/O365-HomeAssistant/pull/105/files#diff-8eb808bbb6b2b32306f7f938d593ca7abca36ad17efc3e237e2aa121dd00e155).

**Comment by RogerSelwyn** [comment]
That is more by accident than design....;-) 'conf' happens to be defined for another purpose earlier in that method. I suspect if you didn't have an email sensor, then it would probably fail. As I said before, fortunately it isn't used by the sensor. 

That said, I've nearly finished coding it up, so that you will have to add it to your config.yaml, otherwise the integration will fail as soon as people upgrade to the version with it in, if people haven't given permission to 'presence.read'.

**Comment by Clem-** [comment]
Right, I saw that the `conf` argument was here because of the copy/paste of the previous line. That's why I removed it in order to make the sensor work.

**Comment by RogerSelwyn** [comment]
Ahh, missed that you removed conf.

Try https://github.com/RogerSelwyn/O365-HomeAssistant/releases/tag/v2.3.0 (beta release). You will need to add to your config.

```
  status_sensors:
    - name: "Roger Teams Status"
```

**Comment by Clem-** [comment]
2.3.0 version seems to require Python 3.8, right?
I'm on 3.7 :/

**Comment by RogerSelwyn** [comment]
Closed with - https://github.com/RogerSelwyn/O365-HomeAssistant/releases/tag/v2.3.0
